lib/send_messages/send_messages_request.js

```diff
@@ -25,10 +25,22 @@ module.exports = function(userId, password, content, recipients, callback) {
 
   recipients.forEach(function(recipient, index) {
     var _recipient = subElement(_recipients, 'ns:recipient');
-    _recipient.attrib.uid = index;
+    if (messageId) {
+      _recipient.attrib.uid = messageId;
+    } else {
+      _recipient.attrib.uid = index;
```

Comment:
Actually it only matters when you want to unschedule a message. Yes the default befaviour is to set index as uid. If there is messageId specified then it will be used for all of them. We can't now easily make it possible to set uid for each recipient. The same problem with other sdks as you know.